staging/src/k8s.io/client-go/util/certificate/certificate_store.go

```diff
@@ -222,6 +225,11 @@ func (s *fileStore) Update(certData, keyData []byte) (*tls.Certificate, error) {
 	if err := s.updateSymlink(certPath); err != nil {
 		return nil, err
 	}
+
+	if err := s.removeObsolete(pemFilename); err != nil {
```

Comment:
Doesn't this break the "rollback" behavior?  The old files are still potentially valid, and moreover the new cert is not guaranteed to work at this point (we haven't tested it).  We shouldn't be deleting the N-1 cert for sure.

Reply:
However , the old cert name is random, no code about rollback found. So I think it is no use.